subworkflows/nf-core/bam_variant_calling_haplotypecaller/main.nf

```diff
@@ -0,0 +1,155 @@
+include { GATK4_HAPLOTYPECALLER  } from '../../../modules/nf-core/gatk4/haplotypecaller/main'
+include { GAWK                   } from '../../../modules/nf-core/gawk/main'
+include { BEDTOOLS_SPLIT         } from '../../../modules/nf-core/bedtools/split/main'
+include { BCFTOOLS_MERGE         } from '../../../modules/nf-core/bcftools/merge/main'
+include { TABIX_TABIX            } from '../../../modules/nf-core/tabix/tabix/main'
+
+workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
+
+    // IMPORTANT: Add the configuration in nextflow.config file to your modules.config
+
+    take:
+    ch_bam              // channel: [mandatory] [meta, bam, bai, bed]
+    ch_scatter          // channel: [optional]  [meta, scatter_count]
+    fasta               // value:   [mandatory] fasta
+    fasta_fai           // value:   [mandatory] fasta_fai
+    dict                // value:   [mandatory] dict
+    ch_dragstr_models   // channel: [optional]  [meta, dragstr_model]
+    dbsnp               // value:   [optional]  dbsnp
+    dbsnp_tbi           // value:   [optional]  dbsnp_tbi
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    //
+    // Optional: Scatter the BED files
+    //
+
+    ch_bam
+        .join(ch_dragstr_models ?: Channel.empty(), remainder: true)
+        .join(ch_scatter ?: Channel.empty(), remainder: true)
+        .branch(
+            { meta, bam, bai, bed, dragstr_model, scatter_count ->
+                new_meta = meta.clone()
+                new_meta.subwf_scatter_count = scatter_count ?: 1
+
+                dragstr = dragstr_model ?: []
+
+                multiple_no_bed: !bed && new_meta.subwf_scatter_count > 1
+                    return [ new_meta, bam, bai, dragstr ]
+                multiple: new_meta.subwf_scatter_count > 1
+                    return [ new_meta, bam, bai, bed, dragstr ]
+                single: new_meta.subwf_scatter_count <= 1
+                    return [ new_meta, bam, bai, bed, dragstr ]
+            }
+        )
+        .set { ch_input }
+
+    GAWK (
+        [[id:'fasta_fai'], fasta_fai],
+        []
+    )
+    ch_versions = ch_versions.mix(GAWK.out.versions)
+
+    ch_input.multiple_no_bed
+        .combine(
+            GAWK.out.output
+                .map(
+                    { meta, bed ->
+                        [ bed ]
+                    }
+                )
+        )
+        .map(
+            { meta, bam, bai, dragstr, bed ->
+                [ meta, bam, bai, bed, dragstr ]
+            }
+        )
+        .mix(ch_input.multiple)
+        .set { ch_to_split }
+
+    BEDTOOLS_SPLIT (
+        ch_to_split.map({ meta, bam, bai, bed, dragstr_model -> [ meta, bed ]}),
+    )
+    ch_versions = ch_versions.mix(BEDTOOLS_SPLIT.out.versions.first())
```

Comment:
should this go in a separate subworkflow? ths would be exactly yhe same for other variantcalling sw

Reply:
👍🏻 love the idea!

Reply:
:exploding_head:

Reply:
OTOH, I _think_ I remember a tool which is able to chunk a bed file into "equal" parts of sorts, which could replace this whole thing... I'm going to have to look it up

Reply:
`bedtools split` already does that :). But I do agree that this should be in a different subworkflow!

Reply:
so https://github.com/nf-core/modules/issues/2540 ?

Reply:
Yeah I'm currently working on it :)

Reply:
https://github.com/nf-core/modules/pull/2544

Reply:
The big question now is: should we add scatter/gather to this subworkflow or let the pipeline developer do this on a pipeline level. 
An argument for not adding it here could be that the pipeline developer could choose which scatter/gather method he/she uses since there will probably be more scatter and gather subworkflows in the future. This way gives more flexibility, but makes this subworkflow quite redundant since it will only run haplotypecaller.

Reply:
I agree! What do you guys think, @maxulysse and @matthdsm? IMO this subworkflow is pretty pointless now since it would only run Haplotypecaller with no channel magic involved